src/Microsoft.ML.Data/Commands/TrainTestCommand.cs

```diff
@@ -188,7 +188,7 @@ private void RunCore(IChannel ch, string cmd)
 
             ILegacyDataLoader testPipe;
             bool hasOutfile = !string.IsNullOrEmpty(ImplOptions.OutputModelFile);
-            var tempFilePath = hasOutfile ? null : Path.GetTempFileName();
+            var tempFilePath = hasOutfile ? null : Path.Combine((Host as IHostEnvironmentInternal).TempFilePath, Path.GetRandomFileName());
```

Comment:
Don't use `as` in these places. Instead just straight cast. That way you get an InvalidCastException when it is wrong, instead of a NullReferenceException.